### DISCUSSION TITLE
extension acting up on Youtube (4k) again.. causing about 10% frame drop

### DISCUSSION THREAD

**Discussion by Sabrehawk** [tamius-han/ultrawidify#255] [Q&A]
Switch the extension off and it goes away...(im not even running fullscreen or anything and it drops 10% with the extension enabled). 
I have run into that problem intermittently with this extension over the years. Dont really know what to do about it.
My firefox settings are fine and i have  powerful machine with a 3070 TI.

**Comment by tamius-han** [comment]
Autodetection is expensive. Try reducing autodetection frequency (or disable it altogether).

——

Aspect ratio autodetection feature works by periodically asking browser to take current video frame and scale it down to a 600ish x 360px frame. This is expensive, and gets more and more expensive as resolution of the video increases. With 4K video, I'm usually looking at near 100 ms just to get the frame. Once the frame is resized to 360p, autodetection itself seems to run in single-digit ms. The numbers suggest that it might be "browser's fault" 

In theory, the operation of getting the frame of the video and rendering it on a  360p canvas is non-blocking. In practice, it's sometimes a bit more blocking than one would think.

Not scaling the video down makes everything _a lot worse_, though — it takes less time for extension to get the current frame, but the time to process that larger frame increases by a lot more than not scaling down the video first saves.
